src/components/ShareStructure/EditShareStructure.vue

```diff
@@ -0,0 +1,490 @@
+    } else if (this.isSeries) {
+      rules.push(v => !(this.shareClasses[this.parentIndex].series
+        .find((s, index) => {
+          // Don't apply uniqueness check to self
+          return index !== this.activeIndex && s.name.split(' Shares')[0].toLowerCase() === v.toLowerCase()
+        })) || 'Series name must be unique')
+      rules.push(v => !(v.split(' ').some(r => this.excludedWordsListForSeries.includes(r.toLowerCase()))) ||
+      'Series name should not contain any of the words share or shares')
+    }
+    return rules
+  }
+
+  private getMaximumShareRule (): Array<Function> {
+    let rules: Array<Function> = []
+    if (!this.hasNoMaximumShares) {
+      rules = [
+        v => !!v || 'Maximum share value is required',
+        v => /^\d+$/.test(v) || 'Must be a number greater than 0']
+      // To prevent changing share class value to a lower value after adding series.
+      if (this.isClass && this.activeIndex !== -1 && !this.hasNoMaximumShares &&
+       this.shareStructure.series.length > 0) {
+        const seriesSum = this.shareStructure.series.reduce((a, b) => +a + +b.maxNumberOfShares, 0)
+        rules.push(v => +v >= seriesSum ||
+        'The number for the series (or all series combined, if there are multiple under ' +
+        'a class) cannot exceed the number for the class')
+      }
+      if (this.isSeries && this.shareClasses[this.parentIndex].hasMaximumShares) {
+        let filteredSeries = this.shareClasses[this.parentIndex].series
+        // The series is in edit mode and should be avoided from the calculating the total
+        if (this.activeIndex !== -1) {
+          filteredSeries = filteredSeries.filter((series) => (
+            series.id !== this.shareClasses[this.parentIndex].series[this.activeIndex].id))
+        }
+        const currentSum = filteredSeries.reduce((a, b) => +a + +b.maxNumberOfShares, 0)
+        rules.push(v => +v + currentSum <= +this.shareClasses[this.parentIndex].maxNumberOfShares ||
+        'The number for the series (or all series combined, if there are multiple under ' +
+        'a class) cannot exceed the number for the class')
+      }
+    }
+    return rules
+  }
+
+  private getParValueRule (): Array<Function> {
+    if (!this.hasNoParValue) {
+      return [
+        v => !!v || 'Par value is required',
+        v => v > 0 || 'Amount must be greater than 0',
+        v => (v < 1) ? (/^(\d+(\.\d{0,3})?|\.\d{0,3})$/.test(v) || 'Amounts less than 1 can be entered with up to 3 decimal place')
+          : (/^\d+(\.\d{1,2})?$/.test(v) || 'Amounts greater than 1 can be entered with up to 2 decimal place')]
+    }
+    return []
+  }
+
+  private getCurrencyRule (): Array<Function> {
+    if (!this.hasNoParValue) {
+      return [v => !!v || 'Currency is required']
+    }
+    return []
+  }
+
+  /** Called when component is created. */
+  private created (): void {
+    if (this.initialValue) {
+      this.shareStructure = { ...this.initialValue }
```

Comment:
If this component is inside a "v-if" in its parent then the initialValue prop is probably set, otherwise I have found that initial props are NOT set until the data is read it (from API), meaning that `created()` has little value in some components. YMMV but maybe worth checking.

Reply:
Will have to look at this a bit more

Reply:
Yeah, just check the initial value in created(). If it's null/undefined then this code does nothing (and your watcher will handle its update).

Reply:
Oh i see. There is definitely a value there when we initiate an 'EDIT' of a share structure or class.

Reply:
OK, I'll leave this with you.